// @nuxt/typescript/dist/typescript.js
/*!
 * @nuxt/typescript v2.8.1 (c) 2016-2019

 * - All the amazing contributors
 * Released under the MIT License.
 * Website: https://nuxtjs.org
*/
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

function _interopDefault (ex) { return (ex && (typeof ex === 'object') && 'default' in ex) ? ex['default'] : ex; }

const fsExtra = require('fs-extra');
const consola = _interopDefault(require('consola'));

const defaultTsJsonConfig = {
  compilerOptions: {
    target: 'esnext',
    module: 'esnext',
    moduleResolution: 'node',
    lib: [
      'esnext',
      'esnext.asynciterable',
      'dom'
    ],
    esModuleInterop: true,
    experimentalDecorators: true,
    allowJs: true,
    sourceMap: true,
    strict: true,
    noImplicitAny: false,
    noEmit: true,
    baseUrl: '.',
    paths: {
      '~/*': [
        './*'
      ],
      '@/*': [
        './*'
      ]
    },
    types: [
      '@types/node',
      '@nuxt/vue-app'
    ]
  }
};

async function setupDefaults(tsConfigPath) {
  let contents = '';

  if (await fsExtra.exists(tsConfigPath)) {
    contents = await fsExtra.readFile(tsConfigPath, 'utf-8');
  }

  if (!contents || contents === '{}') {
    consola.info(`Generating ${tsConfigPath.replace(process.cwd(), '')}`);
    await fsExtra.writeJSON(tsConfigPath, defaultTsJsonConfig, { spaces: 2 });
  }
}

exports.defaultTsJsonConfig = defaultTsJsonConfig;
exports.setupDefaults = setupDefaults;
